src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -495,18 +495,20 @@ private AIState lookForRequests()
                     job.getAsyncRequests().remove(firstDeliverableRequest.getToken());
                 }
 
+                getOwnBuilding().markRequestAsAccepted(worker.getCitizenData(), firstDeliverableRequest.getToken());
+
                 final ItemStack deliveredItemStack = firstDeliverableRequest.getDelivery();
-                if(InventoryUtils.findFirstSlotInItemHandlerWith(
-                        new InvWrapper(worker.getInventoryCitizen()), deliveredItemStack::isItemEqualIgnoreDurability) != -1)
+                if(InventoryUtils.getItemCountInItemHandler(
+                        new InvWrapper(worker.getInventoryCitizen()), deliveredItemStack::isItemEqualIgnoreDurability) >= deliveredItemStack.getCount())
                 {
-                    getOwnBuilding().markRequestAsAccepted(worker.getCitizenData(), firstDeliverableRequest.getToken());
                     return NEEDS_ITEM;
                 }
 
                 //Takes one Stack from the hut if existent
-                if (isInHut(deliveredItemStack)  && !walkToBuilding())
+                if (! walkToBuilding() && InventoryUtils.getItemCountInProvider(getOwnBuilding(), deliveredItemStack::isItemEqualIgnoreDurability) >= deliveredItemStack.getCount() &&
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 182 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

Reply:
OKay this is the only thing which bothers me, can we make it that the worker always walks to the building when he needs something and he waits there for the item?